src/Processors/Formats/Impl/CHColumnToArrowColumn.cpp

```diff
@@ -173,9 +245,71 @@ namespace DB
 
             checkStatus(status, write_column->getName(), format_name);
         }
+    }
 
-        status = builder.Finish(&arrow_array);
-        checkStatus(status, write_column->getName(), format_name);
+    void CHColumnToArrowColumn::fillArrowArray(
+        const String & column_name,
+        ColumnPtr & nested_column,
+        const std::shared_ptr<const IDataType> & column_nested_type,
+        const String column_nested_type_name,
+        std::shared_ptr<arrow::Array> arrow_array,
```

Comment:
I see you removed using `arrow_array` from all places except fillArrowArrayWithDecimalColumnData, why? And I see that `Array(Decimal)` is not supported by now, will you add it? If you need help, ask me.